examples/aggregator/module/rust/src/aggregation.rs

```diff
@@ -0,0 +1,52 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+pub trait Monoid {
+    fn identity() -> Self;
+    fn combine(&self, other: &Self) -> Self;
+}
+
+pub struct Aggregation<T: Monoid> {
+    /// Aggregated data.
+    data: T,
+    /// Number of contributed data samples.
+    sample_number: u64,
+    /// The number of samples that should be collected before for revealing the aggregation.
+    sample_threshold: u64,
+}
+
+impl<T: Monoid + Clone> Aggregation<T> {
+    pub fn new(threshold: u64) -> Self {
+        Aggregation {
+            data: Monoid::identity(),
+            sample_number: 0,
+            sample_threshold: threshold,
+        }
+    }
+
+    pub fn add(&mut self, sample: &T) {
+        self.data = self.data.combine(sample);
+        self.sample_number += 1;
+    }
+
+    pub fn get(&self) -> Option<T> {
```

Comment:
Would `Option<&T>` not work here? If so, I would do that and let the caller do the cloning if necessary, and then you can also drop the `Clone` trait bound from here.